### ISSUE TITLE
Microsoft.WindowsAppRuntime.dll ships with test functions

### ISSUE THREAD

**Issue by riverar** [microsoft/WindowsAppSDK#2345]
### Describe the bug

`Microsoft.WindowsAppRuntime.dll` contains test code that serves no purpose to customers.

```
MddLifetimeManagementTestInitialize
WindowsAppRuntime_SelfContainedTestInitialize
```

### Steps to reproduce the bug

N/A

**Comment by DrusTheAxe** [comment]
ByDesign. These are undocumented, unsupported, small, not harmful merely existing and exist solely for test purposes in ways that required deep internal support to ensure proper testing. This is not unknown in the industry e.g. [SQLite's PRAGMA stats](https://sqlite.org/pragma.html#pragma_stats)

>This pragma returns auxiliary information about tables and indices. The returned information is used during testing to help verify that the query planner is operating correctly. The format and meaning of this pragma will likely change from one release to the next. Because of its volatility, the behavior and output format of this pragma are deliberately undocumented.
>
>The intended use of this pragma is only for testing and validation of SQLite. This pragma is subject to change without notice and is not recommended for use by application programs.

Likewise. They're not APIs. Don't use them. Don't call them. Don't expect stability or even existence.

These aren't the APIs you're looking for :P

**Comment by riverar** [comment]
@DrusTheAxe I'd look the other way if Windows App SDK didn't have a self-contained option. But now that it does, I think we should be taking all efforts to minimize image size, including trimming out useless code. (A self-contained C++ app right now sits around *188MB*.) This code still gets hashed by the loader, its symbols and wil error strings are still baked in, folks still have to download these bytes, etc.

**Comment by jonwis** [comment]
I'm sure there's other ways to accomplish the aims - env.vars come to mind, but then you've got stale env-var query and parsing stuff to figure out. Ideally we're able to test real bits and not just debug bits, but I'm definitely sensitive to the size/space concern.

**Comment by DrusTheAxe** [comment]
Perf, disk size, resource utilization (memory etc), and other factors are considered. Per [SizeBench](https://www.microsoft.com/store/productId/9NDF4N1WG7D6) against a recent Release/x64 build of main

* `MddBootstrapTestInitialize` == 272 bytes
* `Microsoft.WindowsAppRuntime.Bootstrap.dll` == 137,216 bytes

* `WindowsAppRuntime_SelfContainedTestInitialize` == 109 bytes
* `Microsoft.WindowsAppRuntime.dll` == 1,167,360 bytes

A low cost.

I'm not fond of testability in product impactful ways but in certain cases, when necessary, it can be a reasonable price to pay. But it's something we strive to keep to an absolute minimum, for obvious reasons.

**Comment by DrusTheAxe** [comment]
>A self-contained C++ hello world app with WAS right now sits around 188MB

What language and architecture is that?

TL;DR I'm guessing you're looking at a C# project. A C++ hello world app creates a `$(OutputPath)` 100M+ but the actual shipping bits are half that. I'd expect self-contained C# pulls in more (e.g. projections). Does that 188MB also include the `MsixContent` directory? How about the .NETCore runtime?

I don't have a C# app handy at the moment but do have a C++ self-contained app using 1.1-preview1. Taking a closer look...

I created a C++ helloworld-ish app recently in C++ and the x64 **Debug** directory = 116M. But that seems a poor estimate as it the directory has...

* 59.3M of files
* 57 language subdirs for ~1.1M
* Manifests dir = 156K
* Microsoft.UI.Xaml dir = 3K
* A .tlog dir = 59K
* MsixContent dir = 56M

That MsixContent dir appears to be a dupe of the parent. `$(WindowsAppSDKSelfContained)`=true causes the build to unzip the appropriate architecture's framework package's content to the intermediate directory 

```xml
<MicrosoftWindowsAppSDKMsixContent>$(SelfContainedIntermediateOutputPath)MsixContent</MicrosoftWindowsAppSDKMsixContent>
```

That's handled in the nuget's `build\Microsoft.WindowsAppSDK.SelfContained.targets` line 69+

```xml
  <Target Name="GetExtractMicrosoftWindowsAppSDKMsixFilesInputs"
      DependsOnTargets="ResolveReferences">
    <PropertyGroup>
      ...
      <MicrosoftWindowsAppSDKMsixContent>$(SelfContainedIntermediateOutputPath)MsixContent</MicrosoftWindowsAppSDKMsixContent>
...
  <Target Name="ExtractMicrosoftWindowsAppSDKMsixFiles"
      Condition="'$(MSBuildProjectExtension)' != '.wapproj'"
      Inputs="@(MicrosoftWindowsAppSDKMsix)"
      Outputs="$(MicrosoftWindowsAppSDKMsixContent)\Microsoft.UI.Xaml.Controls.pri;$(WindowsAppSDKAppxManifest)"
      DependsOnTargets="GetExtractMicrosoftWindowsAppSDKMsixFilesInputs">

    <Unzip
        SourceFiles="@(MicrosoftWindowsAppSDKMsix)"
        DestinationFolder="$(MicrosoftWindowsAppSDKMsixContent)"
        SkipUnchangedFiles="true"
        OverwriteReadOnlyFiles="true" />
```

That `MsixContent` directory is just intermediary build goo. The actual bits to ship = ~60M. 

Which sounds about right as

```
unzip -v packages\Microsoft.WindowsAppSDK.1.1.0-preview1\tools\MSIX\win10-x64\Microsoft.WindowsAppRuntime.1.1-preview1.msix
```

reports the contents are ~57M.

P.S. the lion's share of that's for code (DLLs). Try

```
unzip -v packages\Microsoft.WindowsAppSDK.1.1.0-preview1\tools\MSIX\win10-x64\Microsoft.WindowsAppRuntime.1.1-preview1.msix *.dll
```

reports *.dll content uncompressed = ~52M. The resources.pri is 1M, the various reasons 1M+ and some miscellaneous files rounds out the total.
